src/content/docs/2/guide/notification.mdx

```diff
@@ -0,0 +1,193 @@
+	npm="https://www.npmjs.com/package/@tauri-apps/plugin-notification"
+	cratesio="https://crates.io/crates/tauri-plugin-notification"
+	jsApi="/2/reference/js/notification"
+	docsrs="https://docs.rs/tauri-plugin-notification/2.0.0-alpha.2/tauri_plugin_notification/"
+/>
+
+Send native notifications to your user using the notification plugin.
+
+{/* TODO: Demo component */}
+
+### Security Considerations
+
+Aside from normal sanitization procedures of user input there are currently no known security considerations.
+
+## Setup
+
+First you have to install the notification plugin in the backend.
+
+<Tabs>
+    <TabItem label="Automatic">
+
+    Use the [add command](/2/reference/cli/#plugin-add) in the Tauri CLI to add the plugin:
+
+    ```bash
+    tauri plugin add notification
+    ```
+
+    {/* TODO: Does this also add it in `lib.rs`? Or does a step need to be added to cover that? */}
+
+    </TabItem>
+    <TabItem label="Manual">
+
+    1. Run `cargo add tauri-plugin-notification` to add the plugin to the project's dependencies in `Cargo.toml`.
+
+    2. Modify `lib.rs` to initialize the plugin:
+
+    ```rust
+    // lib.rs
+    #[cfg_attr(mobile, tauri::mobile_entry_point)]
+    pub fn run() {
+        tauri::Builder::default()
+            // Initialize the plugin
+            .plugin(tauri_plugin_notification::init())
+            .run(tauri::generate_context!())
+            .expect("error while running tauri application");
+    }
+    ```
+
+    If you'd like to use notifications in JavaScript then install the npm package as well:
+
+    <CommandTabs
+        npm="npm install @tauri-apps/plugin-notification"
+        yarn="yarn add @tauri-apps/plugin-notification"
+        pnpm="pnpm add @tauri-apps/plugin-notification"
+    />
+
+    </TabItem>
+
+</Tabs>
+
+## Usage
+
+The notification plugin is available in both JavaScript and Rust.
+
+{/* TODO: Link to which language guide when it's made */}
+
+### Send Notification
+
+<Tabs>
+<TabItem label="JavaScript">
+
+```js
+import {
+	isPermissionGranted,
+	requestPermission,
+	sendNotification,
+} from '@tauri-apps/plugin-notification';
+
+// Do you have permission to send a notification?
+let permissionGranted = await isPermissionGranted();
+
+// If not we need to request it
+if (!permissionGranted) {
+	const permission = await requestPermission();
+	permissionGranted = permission === 'granted';
+}
+
+// Once permission has been granted we can send the notification
+if (permissionGranted) {
+	sendNotification({ title: 'Tauri', body: 'Tauri is awesome!' });
+}
+```
+
+</TabItem>
+<TabItem label="Rust">
+
+Create an instance of `Notification` and then call `.show()` on it:
+
+```rust
+use tauri_plugin_notification::Notification;
+
+#[tauri::command]
+async fn notify(app: AppHandle) -> Result<(), ()> {
+    Notification::new(&app.config().tauri.bundle.identifier)
+        .title("Tauri")
+        .body("Tauri is awesome!")
+        .show()
+}
+```
+
+</TabItem>
+</Tabs>
+
+### Actions
```

Comment:
@simonhyll what do you think about this sort of structure? I think this could go a really long way to give link-able places for support

Reply:
I would put `Security Considerations` further down, feels like the sort of more in-depth topic you read about later on in a post

Not sure what Actions, Attachments or Channels are for? Is it to explain notification concepts or are they general sections that all recipes could have? Options I assume is just a simplified version of what you'd find in the actual API. As long as they aren't too wordy and are copy-paste friendly I'm probably fine with them.

I'm still not overly fond of the terminology of Javascript vs Rust since we should really also have the Rust version avilable using something like Jonas' tauri-sys crate so that the Rust frameworks also get included, because with just JS and Rust (backend) we're kinda leaving them hanging, and I don't see a nice way of adding them without ending up with really annoying names. I'd prefer it if we write about Frontend and Backend as core concepts of Tauri and if people skipped that part then it's not really our problem they refused to read what is almost literally called "the only stuff that's non-optional for you to skip reading". I could also accept Javascript, Rust and Backend I guess because then all frontend languages get included plus one for the backend, but even that isn't completely future proof since we have plans for language bindings at some point.

In the end I defer to your wisdom on the docs in general, so feel free to overrule me on basically anything 😅

Reply:
Note that I read your comments from the bottom to the top and just saw your comment earlier explaining what you meant with Actions, Attachments etc. I'm tired, brain no work 😅

Reply:
Open to whatever you feel would be best for security considerations. I figured if there was anything in-depth that would link out either externally or to a concept and maybe not include it in the guide (unless it really is only very specific to that guide/feature). 

Actions, Attachment, Channels, and Options are part of the plugins APIs, so really just more copy-pasteable examples:

- Actions: https://deploy-preview-1379--tauri-docs-starlight.netlify.app/2/reference/js/notification/#action
- Attachments: https://deploy-preview-1379--tauri-docs-starlight.netlify.app/2/reference/js/notification/#attachment
- Channels: https://deploy-preview-1379--tauri-docs-starlight.netlify.app/2/reference/js/notification/#channel
- Options: https://deploy-preview-1379--tauri-docs-starlight.netlify.app/2/reference/js/notification/#options

RE: Frontend/backend/rust/js how about we put some data-driven decisions in practice and make a poll we can post to Twitter, Discord, and Mastodon and see what comes out? 

> I'm still not overly fond of the terminology of Javascript vs Rust since we should really also have the Rust version avilable using something like Jonas' tauri-sys crate so that the Rust frameworks also get included, because with just JS and Rust (backend) we're kinda leaving them hanging, and I don't see a nice way of adding them without ending up with really annoying names.

Literally have never heard of that until today (looks cool!): https://github.com/JonasKruckenberg/tauri-sys. I do wonder if this is really a more advanced-level usage that's outside of the target audience (i.e. beginners) for docs?

---

A piece that maybe just fell in place for me: do you not like "Rust" because you can have both a Rust frontend (yew, sycamore, etc.) and a Rust backend (application logic, what you're truly referring to as backend) in the same app? Because if that's the case then it just clicked for me and your points are a lot more clear to me now.

Reply:
> Note that I read your comments from the bottom to the top and just saw your comment earlier explaining what you meant with Actions, Attachments etc. I'm tired, brain no work 😅

I mean we've only had like a million comments on this one poor PR the past few days 😅

Reply:
Yea exactly, we want to support our buddies over at the Rust based frontend frameworks, so saying "rust" just for the backend isnt as inclusive

And tauri-sys I would say is the more beginner friendly option than the alternative if you develop using e.g. Yew Leptos or Sycamore. You dont even want to know what the alternative looks like, Jonas really did make aomething awesome and at some point we might wanna make it the actual official approach and "take over" his project. It's kind alike the official JS API but for Rust

Reply:
Could you give some examples of how it's different to use the Rust API in Rust frontend code vs. Rust backend code? Might be better to do it outside of the context of notifications where it's a more practical example, but I think that context is what I'm missing.

Keep in mind that for now we're only covering how to do things with the first-party APIs (I don't think you were thinking of putting `tauri-sys` code in here, and let's talk about it if so, but just wanted to make sure). 

---

I do wonder if "covering all the ways to consume the API" is a similar rabbit hole we got into with "cover all the different JS frontend frameworks"? And if we're wanting to say "here's how you do it in x, then here's a common guide on using Tauri within the context of that". But I doubt we want to NEED to document all of these combinations in every single feature/guide 🤔

Reply:
Even if we dont do tauri-sys specifically skipping Rust entirely will leave our beloved rustaceans behind. Using tauri-sys isnt quite the same thing as the framework rabbit hole imo, but if we skip it we should at least offer a "vanilla Rust" example same as we do with vanilla JS

Reply:
I agree, but we need the dev team to either do that upstream if it doesn't already, or they need to provide the docs and examples to help us write the guides for it. Right now I didn't find any examples for the Rust side of things to be able to fill it out.